apps/chat/src/components/Common/AgentAndToolsetSelector/OverflowButton.tsx

```diff
@@ -0,0 +1,87 @@
+import {
+  autoUpdate,
+  useClick,
+  useDismiss,
+  useFloating,
+  useInteractions,
+} from '@floating-ui/react';
+import { useState } from 'react';
+
+import classNames from 'classnames';
+
+import { MarketplaceEntity } from '@/src/types/marketplace';
+
+import {
+  ESCAPE_KEY_PRESS,
+  OUTSIDE_PRESS_AND_MOUSE_EVENT,
+} from '@/src/constants/modal';
+
+import { OverflowListItem } from './OverflowListItem';
+
+interface OverflowButtonProps {
+  hiddenItems: { id: string; data?: MarketplaceEntity }[];
```

Comment:
guess it should be common component
it's not strictly related to AgentAndToolsetSelector. Lets move the file on 1 level above.

```suggestion
  hiddenItems: { id: string; data?: T }[];
```

also lets add T generic type to be able to reuse it